plugins/native/module/Testing/TestF3DQuakeMDLParser.cxx

```diff
@@ -0,0 +1,279 @@
+#include "vtkF3DQuakeMDLImporter.h"
+
+#include <vtkCallbackCommand.h>
+#include <vtkCommand.h>
+#include <vtkDataSet.h>
+#include <vtkMapper.h>
+#include <vtkMemoryResourceStream.h>
+#include <vtkNew.h>
+#include <vtkRenderer.h>
+#include <vtkTestUtilities.h>
+
+#include <climits>
+#include <iostream>
+#include <regex>
+
+// Header definition : cloned from plugins/native/module/vtkF3DQuakeMDLImporter.cxx
+struct mdl_header_t
+{
+  int IDPO;
+  int version;
+  float scale[3];
+  float translation[3];
+  float boundingRadius;
+  float eyePosition[3];
+  int numSkins;
+  int skinWidth;
+  int skinHeight;
+  int numVertices;
+  int numTriangles;
+  int numFrames;
+  int syncType;
+  int stateFlags;
+  float size;
+};
+
+// Buffer typecasting helper
+template<typename TYPE>
+static TYPE* PeekFromVector(std::vector<uint8_t>& buffer, const size_t& offset)
+{
+  static_assert(std::is_pod<TYPE>::value, "Vector typecast requires POD input");
+
+  if (offset + sizeof(TYPE) > buffer.size())
+  {
+    return nullptr;
+  }
+
+  return reinterpret_cast<TYPE*>(buffer.data() + offset);
+}
+
+struct Error
+{
+  bool parsed = false;
+  std::string fileName = "";
+  std::string lineNumber = "";
+  std::string errorString = "";
+};
+
+Error ErrorCodeParser(const std::string& errorStr)
+{
+  Error err;
+  std::regex errorRegex(R"(ERROR: In (.+?), line (\d+)\n.+: (.+))");
+  std::smatch matches;
+  if (std::regex_search(errorStr, matches, errorRegex))
+  {
+    err.parsed = true;
+    err.fileName = matches[1].str();
+    err.lineNumber = matches[2].str();
+    err.errorString = matches[3].str();
+
+    std::cout << "Parsed error\n    file name : \"" << err.fileName << "\"\n    line number : \""
+              << err.lineNumber << "\"\n    error string : \"" << err.errorString << "\"\n";
+  }
+  return err;
+}
+
+void ErrorCallbackFunc(
+  vtkObject* caller, long unsigned int vtkNotUsed(eventId), void* clientData, void* callData)
+{
+  auto errors = static_cast<std::vector<Error>*>(clientData);
+  errors->push_back(ErrorCodeParser(static_cast<char*>(callData)));
+}
+
+int TestF3DQuakeMDLParser(int vtkNotUsed(argc), char* argv[])
+{
+  // Read in example file, can theoretically be any mdl file
+  std::string path = std::string(argv[1]) + "data/armor.mdl";
+  std::ifstream file(path, std::ios::binary);
+  std::vector<uint8_t> defaultFile(
+    (std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+
+  // setup vectors
+  std::vector<uint8_t> testNoData;
+  std::vector<uint8_t> testNonmagicHeader;
+  std::vector<uint8_t> testNormalHeader;
+  std::vector<uint8_t> testMaxedHeader;
+  std::vector<uint8_t> testMinedHeader;
+
+  // .mdl file with no data
+  {
+    testNoData = {};
+  }
```

Comment:
I'm failing to see what this is testing ?

Reply:
it's not testing anything _here_, there's a series of blocks initializing the data, then another series doing the tests afterwards (including a `stream->SetBuffer(testNoData);`).
Could be clearer to have a block per test case rather group by setup vs run